ecdsa.go

```diff
@@ -300,3 +300,44 @@ func (c *Context) GenerateECDSAKeyPairWithAttributes(public, private AttributeSe
 func (signer *pkcs11PrivateKeyECDSA) Sign(rand io.Reader, digest []byte, opts crypto.SignerOpts) ([]byte, error) {
 	return signer.context.dsaGeneric(signer.handle, pkcs11.CKM_ECDSA, digest)
 }
+
+// Perform CKM_ECDH1_DERIVE function with the given public key
+func (c *Context) ECDH1Derive(pk Signer, pubkey *ecdsa.PublicKey) ([]byte, error) {
+	ecpk := pk.(*pkcs11PrivateKeyECDSA)
```

Comment:
Could you add a checking type assertion here and if `pk` is not an instance of `*pkcs11PrivateKeyECDSA` return an appropriate error.

Reply:
I've done a force push to address these comments.